python-client/giskard/commands/cli_server.py

```diff
@@ -96,6 +96,30 @@ def get_container(version=None, quit_if_not_exists=True) -> Optional[Container]:
             return None
 
 
+def _backend_ready(endpoint) -> bool:
+    try:
+        response = requests.get(endpoint)
+
+        if response.status_code != 200:
+            return False
+
+        return "UP" == response.json()['components']['readinessState']['status']
+    except OSError:
+        return False
+    except KeyError:
+        return False
+
+
+def _await_backend_ready(port: int):
+    endpoint = f"http://localhost:{port}/management/health"
+    backoff_time = 2
+    up = False
+
+    while not up:
```

Comment:
I think we should have a waiting limit. Let's say we wait for 3 minutes and if `_backend_ready` still returns False we break the loop and print an explanation like:

```
Giskard backend takes unusually long time to start, please check the logs with `giskard server logs backend`
```

Also I think it's worth adding a feedback while waiting, for example printing dots every `backoff_time` seconds, like this it's clear that the process isn't frozen, but is actually checking the status
```
click.echo(".", nl=False)
```
should do the trick, because print(end='') won't work in the loop without explicit stdout flushing